docs/docs/08-guides/03-react-vite-websockets.md

```diff
@@ -0,0 +1,492 @@
+---
+title: React, Vite & Websockets 
+id: react-vite-websockets
+keywords: [Websockets, React, Vite, Local, Wing]
+---
+
+# React, Vite & Websockets 
+
+In this guide, we will go through the process of building a simple React webpage that connects to a Wing backend.
+
+The webpage will showcase a counter that can be incremented by clicking on it. This counter will increment a distributed counter deployed via the Wing backend. We will also demonstrate how to get real-time updates from the backend to the browser every time the counter is updated from any other webpage, using Websockets.
+
+## Prerequisites
+
+- Node.js v20 or later.
+- Wing supports two popular IDEs with syntax highlighting, completions, go-to-definition and an embedded Wing Console:
+
+  - [VSCode](https://marketplace.visualstudio.com/items?itemName=Monada.vscode-wing) - Official extension
+  - [IntelliJ](https://plugins.jetbrains.com/plugin/22353-wing) - Community extension
+
+## Step 1 - Installation & Scaffolding
+In this step, we will be creating our project by setting up two folders:
+
+ - `backend` - The place for Wing's backend code
+ - `frontend` - The place for our React app code
+
+### Creating a React App with Vite
+
+1. Create the root project folder:
+ ```sh
+ mkdir ~/shared-counter
+ cd ~/shared-counter
+ ```
+2. Create a new React app using Vite under the frontend folder:
+ ```sh
+ npm create vite frontend -- --template react-ts
+ ```
+3. Run your React app and ensure it works:
+ ```sh
+ cd frontend
+ npm install
+ npm run dev
+ ```
+> The result should be a very simple webpage that runs locally and works without any backend (note the counter is not shared between tabs).
+
+4. Press Ctrl-C to return to the CLI prompt.
+
+### Creating a Wing backend
+
+1. Install Wing:
+ ```sh
+ npm install -g winglang
+ # Verify installation
+ wing -V 
+ ```
+2. Create a `backend` directory under the project root:
+ ```sh
+ mkdir ~/shared-counter/backend
+ cd ~/shared-counter/backend
+ ```
+3. Generate a new empty Wing project:
+ ```sh
+ wing new empty
+ ```
+> This will generate three files: `package.json`, `package-lock.json` and `main.w` file with a simple hello-world `Cloud.Function`
+4. Run this project inside Wing Simulator
+```sh
+wing run main.w
+```
+> The result should be a page shows a single `Cloud.Function` if you invoke it, it should show `hello, world` in the response section.
+5. Ctrl-C to go back to CLI prompt.
```

Comment:
don't you want them to interact with the console - i see that it is just to click on the `invoke` button... should be easy and cool